config/rbac/role.yaml

```diff
@@ -42,6 +42,18 @@ rules:
   - get
   - list
   - watch
+- apiGroups:
+  - networking.k8s.io
```

Comment:
Not directly related to this line but, I think something is missing for OpenShift Routes: 
Did you try to deploy the Tempo Operator in OpenShift? Maybe the `routes/custom-host` resource is missing.

[More information here](https://github.com/open-telemetry/opentelemetry-operator/pull/1337)

Reply:
Yes, I tried it on OpenShift and it works (tested with crc). Pavol added the `route.openshift.io` RBAC as part of the gateway PRs.
I think `routes/custom-host` from the linked PR is only required if we set a custom hostname for the route (which is not yet possible in this PR).

Currently I've both a `ingress:` and `route:` field in the spec. Should I change it to how otel-operator has it with a single `ingress` field in the spec, and a `type` field which can toggle Ingress vs. Route?

Reply:
My original idea why I created two separate fields for Route and Ingress that they may have different fields, for example some option is only available for Routes (e.g. custom TLS settings) and not for Ingress and vice-versa. Some are possible in both objects though, like hostname and annotations.

Reply:
I see. Well, it totally makes sense in that case.